integration_tests/helpers/configHelpers.ts

```diff
@@ -0,0 +1,61 @@
+import getCsvConverter from "csvtojson";
+import { getS3Object } from "./s3Helper";
+import { configStackName } from "./envHelper";
+
+interface VendorServiceConfigRow {
+  vendor_name: string;
+  vendor_regex: string;
+  client_id: string;
+  service_name: string;
+  service_regex: string;
+  event_name: string;
+}
+
+let cachedVendorServiceConfig: VendorServiceConfigRow[] | undefined;
+
+export const getVendorServiceConfigRow = async (
+  fields: Partial<VendorServiceConfigRow>
+): Promise<VendorServiceConfigRow> => {
+  if (cachedVendorServiceConfig === undefined) {
+    const vendorServiceConfigText = await getS3Object({
+      bucket: configStackName(),
+      key: "vendor_services/vendor-services.csv",
+    });
+
+    if (vendorServiceConfigText === undefined)
+      throw Error("No vendor service config found");
+
+    const csvConverter = getCsvConverter();
+
+    const vendorServiceConfig = await csvConverter.fromString(
+      vendorServiceConfigText
+    );
+
+    if (!vendorServiceConfig.every(isVendorServiceConfigRow))
+      throw new Error("Invalid vendor service config");
+
+    cachedVendorServiceConfig = vendorServiceConfig;
+  }
+
+  const vendorServiceConfigRow = cachedVendorServiceConfig.find((row) =>
+    Object.entries(fields).every(
+      ([fieldName, fieldValue]) =>
+        fieldValue === undefined ||
+        row[fieldName as keyof VendorServiceConfigRow] === fieldValue
+    )
+  );
+
+  if (vendorServiceConfigRow === undefined)
+    throw Error("Vendor service config row not found");
+
+  return vendorServiceConfigRow;
+};
```

Comment:
Just a suggestion, not necessary, but you could split this out to reduce nesting and avoid reassigning `cachedVendorServiceConfig` like this 
```
const cachedVendorServiceConfig = (async (): Promise<
  VendorServiceConfigRow[]
> => {
  const vendorServiceConfigText = await getS3Object({
    bucket: configStackName(),
    key: "vendor_services/vendor-services.csv",
  });

  if (vendorServiceConfigText === undefined)
    throw Error("No vendor service config found");

  const csvConverter = getCsvConverter();

  const vendorServiceConfig = await csvConverter.fromString(
    vendorServiceConfigText
  );

  if (!vendorServiceConfig.every(isVendorServiceConfigRow))
    throw new Error("Invalid vendor service config");

  return vendorServiceConfig;
})();

export const getVendorServiceConfigRow = async (
  fields: Partial<VendorServiceConfigRow>
): Promise<VendorServiceConfigRow> => {
  const vendorServiceConfig = await cachedVendorServiceConfig;

  const vendorServiceConfigRow = vendorServiceConfig.find((row) =>
    Object.entries(fields).every(
      ([fieldName, fieldValue]) =>
        fieldValue === undefined ||
        row[fieldName as keyof VendorServiceConfigRow] === fieldValue
    )
  );

  if (vendorServiceConfigRow === undefined)
    throw Error("Vendor service config row not found");

  return vendorServiceConfigRow;
};
```

That self invocation is a bit freaky so here's a model for this I played around with in a terminal to assure myself that `getS3Object` wouldn't be called over and over 
```
const slow = async (arg) =>
  new Promise((resolve) => {
    console.log('new underlying promise spawned')
    setTimeout(() => {
      resolve(`You sent me ${arg}`);
    }, 1000);
  });

const data = (async () => {
  const res = await slow('thing');
  return res;
})();

(async () => {
    const a = await data;
    const b = await data;
    const c = await data;
    console.log(a,b,c)
})()
```